src/containers/auth/register/Register.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { Button } from "react-bootstrap";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+
+import { useAuth } from "../AuthProvider";
+
+import { USER } from "../../../helpers/auth";
+
+const Register = () => {
+  const initialValues = {
+    username: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const { registerUserFlow } = useAuth();
+
+  const validationSchema = Yup.object({
+    username: Yup.string()
+      .required(USER.USER_NAME_REQUIRED)
+      .min(3, USER.USER_NAME_VALIDATION),
+    password: Yup.string()
+      .required(USER.PASSWORD_REQUIRED)
+      .min(6, USER.PASSWORD_VALIDATIOM),
+    confirmPassword: Yup.string() // Validate password confirmation
+      .oneOf([Yup.ref(USER.PASSWORD), null], USER.PASSWORD_MATCH),
+  });
+
+  const onSubmit = async (values, { resetForm }) => {
+    const user = values.username;
+    const pwd = values.password;
+    registerUserFlow(user, pwd);
+    resetForm();
+  };
+
+  return (
+    <main className="App">
+      <section>
+        <h2>Register</h2>
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={onSubmit}
+        >
+          <Form>
+            <div>
+              <label htmlFor="username">Username</label>
+              <Field
+                type="text"
+                id="username"
+                name="username"
+                className="fieldWrapper"
+              />
+              <ErrorMessage name="username" component="div" className="error" />
+            </div>
+
+            <div>
+              <label htmlFor="password">Password</label>
+              <Field
+                type="password"
+                id="password"
+                name="password"
+                className="fieldWrapper"
+              />
+              <ErrorMessage name="password" component="div" className="error" />
+            </div>
+            <div>
+              <label htmlFor="confirmPassword">Confirm Password</label>
+              <Field
+                type="password"
+                id="confirmPassword"
+                name="confirmPassword"
+                className="fieldWrapper"
+              />
+              <ErrorMessage
+                name="confirmPassword"
+                component="div"
+                className="error"
+              />
+            </div>
+            <Button type="submit" variant="light" data-testid="signup-button">
```

Comment:
Lets follow the test element naming convention https://empyreallogistics.atlassian.net/wiki/spaces/E/pages/153518081/Prerequisites+and+Guidelines#Test-Elements

Reply:
Fixed

Reply:
test elements id should be updated

register-signup-button -> register-signup_button
signin-span -> register-signin_span

For reference - https://empyreallogistics.atlassian.net/wiki/spaces/E/pages/153518081/Prerequisites+and+Guidelines#Test-Elements

Reply:
Fixed